File: src/1-100/20. Valid Parentheses/Solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        mapping = {')': '(', '}': '{', ']': '['}

        for c in s:
            if c in mapping:
                if mapping[c] != (stack.pop() if stack else ''):
                    return False
            else:
                stack.append(c)

        return not stack

File: src/1-100/20. Valid Parentheses/Solution.js
/**
 * @param { string } s
 * @return { boolean }
 */
const isValid = function (s) {
	const stack = []
	const mapping = { ')': '(', '}': '{', ']': '[' }

	for (const c of s.split('')) {
		if (c in mapping) {
			if (mapping[c] !== (stack.length > 0 ? stack.pop() : '')) return false
		} else {
			stack.push(c)
		}
	}

	return stack.length === 0
}
